src/OrbitGl/App.cpp

```diff
@@ -1198,6 +1189,53 @@ orbit_base::Future<ErrorMessageOr<std::filesystem::path>> OrbitApp::RetrieveModu
   return final_result;
 }
 
+orbit_base::Future<ErrorMessageOr<std::filesystem::path>> OrbitApp::RetrieveModuleWithDebugInfo(
+    const ModuleData* module_data) {
+  return RetrieveModuleWithDebugInfo(module_data->file_path(), module_data->build_id());
+}
+
+orbit_base::Future<ErrorMessageOr<std::filesystem::path>> OrbitApp::RetrieveModuleWithDebugInfo(
+    const std::string& module_path, const std::string& build_id) {
+  auto loaded_module = RetrieveModule(module_path, build_id);
+  return loaded_module.ThenIfSuccess(
+      main_thread_executor_,
+      [this, module_path](
+          const std::filesystem::path& local_file_path) -> ErrorMessageOr<std::filesystem::path> {
+        auto elf_file = orbit_elf_utils::ElfFile::Create(local_file_path);
+
+        if (elf_file.has_error()) return elf_file.error();
```

Comment:
OUTCOME_TRY(elf_file, orbit_elf_utils::ElfFile::Create(local_file_path));

Reply:
I'd rather not use OUTCOME_TRY before we haven't figured out if we move to absl::StatusOr